storey/flow.py

```diff
@@ -531,20 +531,29 @@ def __init__(self, max_in_flight=8, **kwargs):
         self._pending_by_key = {}
 
     async def _worker(self):
+        event = None
+        received_job_count = 0
+        self_sent_jobs = {}
         try:
             while True:
-                job = await self._q.get()
-                if job is _termination_obj:
-                    for pending_event in self._pending_by_key.values():
-                        if pending_event.pending and not pending_event.in_flight:
-                            resp = await self._process_event(pending_event.pending)
-                            for event in pending_event.pending:
-                                await self._handle_completed(event, resp)
-                    if self._q.empty():
+                jobs = self_sent_jobs.pop(received_job_count, None)
+                if jobs:
+                    job = jobs[0]
+                    if len(jobs) > 1:
+                        self_sent_jobs[received_job_count] = jobs[1:]
+                else:
+                    job = await self._q.get()
+                    received_job_count += 1
+                    if job is _termination_obj:
+                        if received_job_count in self_sent_jobs:
```

Comment:
Why would `received_job_count in self_sent_jobs` ? you are checking it at L539 and nothing else but this function isn’t adding to `self_sent_jobs` so it should not exist there, no?